rosserial_arduino/src/rosserial_arduino/make_libraries_isolated.py

```diff
@@ -89,6 +89,11 @@
 microcontroller_ws = parsed_contents["microcontroller_ws"]
 header_out = os.path.join(microcontroller_ws, parsed_contents["header_output_directory"])
 source_out = os.path.join(microcontroller_ws, parsed_contents["source_output_directory"])
+
+microcontroller_ws = os.path.expanduser(microcontroller_ws)
+header_out = os.path.expanduser(header_out)
+source_out = os.path.expanduser(source_out)
```

Comment:
I don't think `header_out` and `source_out` need `os.path.expanduser`. They are paths relative to `microcontroller_ws`, not absolute paths.